src/receive_imf.rs

```diff
@@ -1454,12 +1454,18 @@ async fn add_parts(
                     let relay_server = node_addr.relay_url().map(|relay| relay.as_str());
                     iroh_add_peer_for_topic(context, instance_id, topic, node_id, relay_server)
                         .await?;
-                    let iroh = context.get_or_try_init_peer_channel().await?;
-                    iroh.maybe_add_gossip_peers(topic, vec![node_addr]).await?;
+                    if context
```

Comment:
I would if we have this check, do it even further up in code to not parse nodeAddr etc.
If you just enabled realtime API you will not know any peeers, but think its fine because will advertise yourself.